// @asposecloud/aspose-ocr-cloud/dist/model/OCRRegion.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _OCRRect = _interopRequireDefault(require("./OCRRect"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The OCRRegion model module.
 * @module model/OCRRegion
 * @version 5.0
 */
var OCRRegion = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OCRRegion</code>.
   * Represents information about strict regions to recognize text
   * @alias module:model/OCRRegion
   */
  function OCRRegion() {
    _classCallCheck(this, OCRRegion);
    OCRRegion.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(OCRRegion, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>OCRRegion</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCRRegion} obj Optional instance to populate.
     * @return {module:model/OCRRegion} The populated <code>OCRRegion</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OCRRegion();
        if (data.hasOwnProperty('rect')) {
          obj['rect'] = _OCRRect["default"].constructFromObject(data['rect']);
        }
        if (data.hasOwnProperty('order')) {
          obj['order'] = _ApiClient["default"].convertToType(data['order'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>OCRRegion</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCRRegion</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `rect`
      if (data['rect']) {
        // data not null
        _OCRRect["default"].validateJSON(data['rect']);
      }
      return true;
    }
  }]);
  return OCRRegion;
}();
/**
 * @member {module:model/OCRRect} rect
 */
OCRRegion.prototype['rect'] = undefined;

/**
 * The serial number of the region for the formation of the text
 * @member {Number} order
 */
OCRRegion.prototype['order'] = undefined;
var _default = exports["default"] = OCRRegion;
