// @asposecloud/aspose-ocr-cloud/dist/model/OCRSettingsRecognizeFont.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DsrConfidence = _interopRequireDefault(require("./DsrConfidence"));
var _DsrMode = _interopRequireDefault(require("./DsrMode"));
var _Language = _interopRequireDefault(require("./Language"));
var _OCRRegion = _interopRequireDefault(require("./OCRRegion"));
var _ResultType = _interopRequireDefault(require("./ResultType"));
var _ResultTypeTable = _interopRequireDefault(require("./ResultTypeTable"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The OCRSettingsRecognizeFont model module.
 * @module model/OCRSettingsRecognizeFont
 * @version 5.0
 */
var OCRSettingsRecognizeFont = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OCRSettingsRecognizeFont</code>.
   * @alias module:model/OCRSettingsRecognizeFont
   */
  function OCRSettingsRecognizeFont() {
    _classCallCheck(this, OCRSettingsRecognizeFont);
    OCRSettingsRecognizeFont.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(OCRSettingsRecognizeFont, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>OCRSettingsRecognizeFont</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCRSettingsRecognizeFont} obj Optional instance to populate.
     * @return {module:model/OCRSettingsRecognizeFont} The populated <code>OCRSettingsRecognizeFont</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OCRSettingsRecognizeFont();
        if (data.hasOwnProperty('language')) {
          obj['language'] = _Language["default"].constructFromObject(data['language']);
        }
        if (data.hasOwnProperty('makeSkewCorrect')) {
          obj['makeSkewCorrect'] = _ApiClient["default"].convertToType(data['makeSkewCorrect'], 'Boolean');
        }
        if (data.hasOwnProperty('makeSpellCheck')) {
          obj['makeSpellCheck'] = _ApiClient["default"].convertToType(data['makeSpellCheck'], 'Boolean');
        }
        if (data.hasOwnProperty('makeContrastCorrection')) {
          obj['makeContrastCorrection'] = _ApiClient["default"].convertToType(data['makeContrastCorrection'], 'Boolean');
        }
        if (data.hasOwnProperty('resultType')) {
          obj['resultType'] = _ResultType["default"].constructFromObject(data['resultType']);
        }
        if (data.hasOwnProperty('Rotate')) {
          obj['Rotate'] = _ApiClient["default"].convertToType(data['Rotate'], 'Number');
        }
        if (data.hasOwnProperty('makeBinarization')) {
          obj['makeBinarization'] = _ApiClient["default"].convertToType(data['makeBinarization'], 'Boolean');
        }
        if (data.hasOwnProperty('makeUpsampling')) {
          obj['makeUpsampling'] = _ApiClient["default"].convertToType(data['makeUpsampling'], 'Boolean');
        }
        if (data.hasOwnProperty('dsrMode')) {
          obj['dsrMode'] = _DsrMode["default"].constructFromObject(data['dsrMode']);
        }
        if (data.hasOwnProperty('dsrConfidence')) {
          obj['dsrConfidence'] = _DsrConfidence["default"].constructFromObject(data['dsrConfidence']);
        }
        if (data.hasOwnProperty('resultTypeTable')) {
          obj['resultTypeTable'] = _ResultTypeTable["default"].constructFromObject(data['resultTypeTable']);
        }
        if (data.hasOwnProperty('regions')) {
          obj['regions'] = _ApiClient["default"].convertToType(data['regions'], [_OCRRegion["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>OCRSettingsRecognizeFont</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCRSettingsRecognizeFont</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['regions']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['regions'])) {
          throw new Error("Expected the field `regions` to be an array in the JSON data but got " + data['regions']);
        }
        // validate the optional field `regions` (array)
        var _iterator = _createForOfIteratorHelper(data['regions']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _OCRRegion["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      return true;
    }
  }]);
  return OCRSettingsRecognizeFont;
}();
/**
 * @member {module:model/Language} language
 */
OCRSettingsRecognizeFont.prototype['language'] = undefined;

/**
 * @member {Boolean} makeSkewCorrect
 * @default true
 */
OCRSettingsRecognizeFont.prototype['makeSkewCorrect'] = true;

/**
 * @member {Boolean} makeSpellCheck
 * @default false
 */
OCRSettingsRecognizeFont.prototype['makeSpellCheck'] = false;

/**
 * @member {Boolean} makeContrastCorrection
 * @default false
 */
OCRSettingsRecognizeFont.prototype['makeContrastCorrection'] = false;

/**
 * @member {module:model/ResultType} resultType
 */
OCRSettingsRecognizeFont.prototype['resultType'] = undefined;

/**
 * @member {Number} Rotate
 */
OCRSettingsRecognizeFont.prototype['Rotate'] = undefined;

/**
 * @member {Boolean} makeBinarization
 * @default true
 */
OCRSettingsRecognizeFont.prototype['makeBinarization'] = true;

/**
 * Option to enable image up-sampling algorithm to improve quality. True by default
 * @member {Boolean} makeUpsampling
 * @default false
 */
OCRSettingsRecognizeFont.prototype['makeUpsampling'] = false;

/**
 * @member {module:model/DsrMode} dsrMode
 */
OCRSettingsRecognizeFont.prototype['dsrMode'] = undefined;

/**
 * @member {module:model/DsrConfidence} dsrConfidence
 */
OCRSettingsRecognizeFont.prototype['dsrConfidence'] = undefined;

/**
 * @member {module:model/ResultTypeTable} resultTypeTable
 */
OCRSettingsRecognizeFont.prototype['resultTypeTable'] = undefined;

/**
 * @member {Array.<module:model/OCRRegion>} regions
 */
OCRSettingsRecognizeFont.prototype['regions'] = undefined;
var _default = exports["default"] = OCRSettingsRecognizeFont;
