// @asposecloud/aspose-ocr-cloud/dist/model/TTSResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ResponseStatusCode = _interopRequireDefault(require("./ResponseStatusCode"));
var _TTSError = _interopRequireDefault(require("./TTSError"));
var _TTSResult = _interopRequireDefault(require("./TTSResult"));
var _TTSTaskStatus = _interopRequireDefault(require("./TTSTaskStatus"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The TTSResponse model module.
 * @module model/TTSResponse
 * @version 5.0
 */
var TTSResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TTSResponse</code>.
   * @alias module:model/TTSResponse
   */
  function TTSResponse() {
    _classCallCheck(this, TTSResponse);
    TTSResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(TTSResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>TTSResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TTSResponse} obj Optional instance to populate.
     * @return {module:model/TTSResponse} The populated <code>TTSResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TTSResponse();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('responseStatusCode')) {
          obj['responseStatusCode'] = _ResponseStatusCode["default"].constructFromObject(data['responseStatusCode']);
        }
        if (data.hasOwnProperty('taskStatus')) {
          obj['taskStatus'] = _TTSTaskStatus["default"].constructFromObject(data['taskStatus']);
        }
        if (data.hasOwnProperty('results')) {
          obj['results'] = _ApiClient["default"].convertToType(data['results'], [_TTSResult["default"]]);
        }
        if (data.hasOwnProperty('error')) {
          obj['error'] = _TTSError["default"].constructFromObject(data['error']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>TTSResponse</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TTSResponse</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      if (data['results']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['results'])) {
          throw new Error("Expected the field `results` to be an array in the JSON data but got " + data['results']);
        }
        // validate the optional field `results` (array)
        var _iterator = _createForOfIteratorHelper(data['results']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _TTSResult["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // validate the optional field `error`
      if (data['error']) {
        // data not null
        _TTSError["default"].validateJSON(data['error']);
      }
      return true;
    }
  }]);
  return TTSResponse;
}();
/**
 * @member {String} id
 */
TTSResponse.prototype['id'] = undefined;

/**
 * @member {module:model/ResponseStatusCode} responseStatusCode
 */
TTSResponse.prototype['responseStatusCode'] = undefined;

/**
 * @member {module:model/TTSTaskStatus} taskStatus
 */
TTSResponse.prototype['taskStatus'] = undefined;

/**
 * @member {Array.<module:model/TTSResult>} results
 */
TTSResponse.prototype['results'] = undefined;

/**
 * @member {module:model/TTSError} error
 */
TTSResponse.prototype['error'] = undefined;
var _default = exports["default"] = TTSResponse;
