// @asposecloud/aspose-ocr-cloud/dist/api/DewarpImageApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _OCRDewarpImageBody = _interopRequireDefault(require("../model/OCRDewarpImageBody"));
var _OCRResponse = _interopRequireDefault(require("../model/OCRResponse"));
var _ProblemDetails = _interopRequireDefault(require("../model/ProblemDetails"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* DewarpImage service.
* @module api/DewarpImageApi
* @version 5.0
*/
var DewarpImageApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new DewarpImageApi. 
  * @alias module:api/DewarpImageApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function DewarpImageApi(apiClient) {
    _classCallCheck(this, DewarpImageApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the cancelDewarpImage operation.
   * @callback module:api/DewarpImageApi~cancelDewarpImageCallback
   * @param {String} error Error message, if any.
   * @param data This operation does not return a value.
   * @param {String} response The complete HTTP response.
   */

  /**
   * CancelDewarpImage
   * @param {String} id 
   * @param {module:api/DewarpImageApi~cancelDewarpImageCallback} callback The callback function, accepting three arguments: error, data, response
   */
  _createClass(DewarpImageApi, [{
    key: "cancelDewarpImage",
    value: function cancelDewarpImage(id, callback) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelDewarpImage");
      }
      var pathParams = {};
      var queryParams = {
        'id': id
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = [];
      var accepts = [];
      var returnType = null;
      return this.apiClient.callApi('/v5.0/ocr/DewarpImage', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getDewarpImage operation.
     * @callback module:api/DewarpImageApi~getDewarpImageCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetDewarpImage
     * @param {String} id 
     * @param {module:api/DewarpImageApi~getDewarpImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
  }, {
    key: "getDewarpImage",
    value: function getDewarpImage(id, callback) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getDewarpImage");
      }
      var pathParams = {};
      var queryParams = {
        'id': id
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _OCRResponse["default"];
      return this.apiClient.callApi('/v5.0/ocr/DewarpImage', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the postDewarpImage operation.
     * @callback module:api/DewarpImageApi~postDewarpImageCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostDewarpImage
     * @param {module:model/OCRDewarpImageBody} oCRDewarpImageBody 
     * @param {module:api/DewarpImageApi~postDewarpImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
  }, {
    key: "postDewarpImage",
    value: function postDewarpImage(oCRDewarpImageBody, callback) {
      var postBody = oCRDewarpImageBody;
      // verify the required parameter 'oCRDewarpImageBody' is set
      if (oCRDewarpImageBody === undefined || oCRDewarpImageBody === null) {
        throw new Error("Missing the required parameter 'oCRDewarpImageBody' when calling postDewarpImage");
      }
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = ['application/json'];
      var accepts = ['text/plain'];
      var returnType = 'String';
      return this.apiClient.callApi('/v5.0/ocr/DewarpImage', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
  return DewarpImageApi;
}();
