// @asposecloud/aspose-ocr-cloud/dist/api/ImageProcessingApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _OCRResponse = _interopRequireDefault(require("../model/OCRResponse"));
var _ProblemDetails = _interopRequireDefault(require("../model/ProblemDetails"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* ImageProcessing service.
* @module api/ImageProcessingApi
* @version 5.0
*/
var ImageProcessingApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new ImageProcessingApi. 
  * @alias module:api/ImageProcessingApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function ImageProcessingApi(apiClient) {
    _classCallCheck(this, ImageProcessingApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the getResultFile operation.
   * @callback module:api/ImageProcessingApi~getResultFileCallback
   * @param {String} error Error message, if any.
   * @param {Object} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * GetResultFile
   * @param {String} id 
   * @param {module:api/ImageProcessingApi~getResultFileCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link Object}
   */
  _createClass(ImageProcessingApi, [{
    key: "getResultFile",
    value: function getResultFile(id, callback) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getResultFile");
      }
      var pathParams = {};
      var queryParams = {
        'id': id
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = [];
      var accepts = ['image/png'];
      var returnType = Object;
      return this.apiClient.callApi('/v5.0/ocr/ImageProcessing/GetResultFile', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getResultTask operation.
     * @callback module:api/ImageProcessingApi~getResultTaskCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetResultTask
     * @param {String} id 
     * @param {module:api/ImageProcessingApi~getResultTaskCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
  }, {
    key: "getResultTask",
    value: function getResultTask(id, callback) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getResultTask");
      }
      var pathParams = {};
      var queryParams = {
        'id': id
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _OCRResponse["default"];
      return this.apiClient.callApi('/v5.0/ocr/ImageProcessing/GetResultTask', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the postBinarizationFile operation.
     * @callback module:api/ImageProcessingApi~postBinarizationFileCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostBinarizationFile
     * @param {Object} opts Optional parameters
     * @param {File} opts.file 
     * @param {module:api/ImageProcessingApi~postBinarizationFileCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
  }, {
    key: "postBinarizationFile",
    value: function postBinarizationFile(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'file': opts['file']
      };
      var authNames = ['JWT'];
      var contentTypes = ['multipart/form-data'];
      var accepts = ['text/plain'];
      var returnType = 'String';
      return this.apiClient.callApi('/v5.0/ocr/ImageProcessing/PostBinarizationFile', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the postDewarpingFile operation.
     * @callback module:api/ImageProcessingApi~postDewarpingFileCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostDewarpingFile
     * @param {Object} opts Optional parameters
     * @param {File} opts.file 
     * @param {module:api/ImageProcessingApi~postDewarpingFileCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
  }, {
    key: "postDewarpingFile",
    value: function postDewarpingFile(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'file': opts['file']
      };
      var authNames = ['JWT'];
      var contentTypes = ['multipart/form-data'];
      var accepts = ['text/plain'];
      var returnType = 'String';
      return this.apiClient.callApi('/v5.0/ocr/ImageProcessing/PostDewarpingFile', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the postSkewCorrectionFile operation.
     * @callback module:api/ImageProcessingApi~postSkewCorrectionFileCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostSkewCorrectionFile
     * @param {Object} opts Optional parameters
     * @param {File} opts.file 
     * @param {module:api/ImageProcessingApi~postSkewCorrectionFileCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
  }, {
    key: "postSkewCorrectionFile",
    value: function postSkewCorrectionFile(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'file': opts['file']
      };
      var authNames = ['JWT'];
      var contentTypes = ['multipart/form-data'];
      var accepts = ['text/plain'];
      var returnType = 'String';
      return this.apiClient.callApi('/v5.0/ocr/ImageProcessing/PostSkewCorrectionFile', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the postUpsamplingFile operation.
     * @callback module:api/ImageProcessingApi~postUpsamplingFileCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostUpsamplingImageFile
     * @param {Object} opts Optional parameters
     * @param {File} opts.file 
     * @param {module:api/ImageProcessingApi~postUpsamplingFileCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
  }, {
    key: "postUpsamplingFile",
    value: function postUpsamplingFile(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'file': opts['file']
      };
      var authNames = ['JWT'];
      var contentTypes = ['multipart/form-data'];
      var accepts = ['text/plain'];
      var returnType = 'String';
      return this.apiClient.callApi('/v5.0/ocr/ImageProcessing/PostUpsamplingImageFile', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
  return ImageProcessingApi;
}();
