// @asposecloud/aspose-ocr-cloud/dist/api/TextToSpeechApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ProblemDetails = _interopRequireDefault(require("../model/ProblemDetails"));
var _TTSBodyDeprecated = _interopRequireDefault(require("../model/TTSBodyDeprecated"));
var _TTSResponse = _interopRequireDefault(require("../model/TTSResponse"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* TextToSpeech service.
* @module api/TextToSpeechApi
* @version 5.0
*/
var TextToSpeechApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new TextToSpeechApi. 
  * @alias module:api/TextToSpeechApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function TextToSpeechApi(apiClient) {
    _classCallCheck(this, TextToSpeechApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the getTextToSpeechResult operation.
   * @callback module:api/TextToSpeechApi~getTextToSpeechResultCallback
   * @param {String} error Error message, if any.
   * @param {module:model/TTSResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * GetTextToSpeechResult
   * @param {String} id 
   * @param {module:api/TextToSpeechApi~getTextToSpeechResultCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/TTSResponse}
   */
  _createClass(TextToSpeechApi, [{
    key: "getTextToSpeechResult",
    value: function getTextToSpeechResult(id, callback) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getTextToSpeechResult");
      }
      var pathParams = {};
      var queryParams = {
        'id': id
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _TTSResponse["default"];
      return this.apiClient.callApi('/v5.0/ocr/TextToSpeech/GetTextToSpeechResult', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getTextToSpeechResultFile operation.
     * @callback module:api/TextToSpeechApi~getTextToSpeechResultFileCallback
     * @param {String} error Error message, if any.
     * @param {Object} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetTextToSpeechResultFile
     * @param {String} id 
     * @param {module:api/TextToSpeechApi~getTextToSpeechResultFileCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Object}
     */
  }, {
    key: "getTextToSpeechResultFile",
    value: function getTextToSpeechResultFile(id, callback) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getTextToSpeechResultFile");
      }
      var pathParams = {};
      var queryParams = {
        'id': id
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = [];
      var accepts = ['application/octet-stream'];
      var returnType = Object;
      return this.apiClient.callApi('/v5.0/ocr/TextToSpeech/GetTextToSpeechResultFile', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the postTextToSpeech operation.
     * @callback module:api/TextToSpeechApi~postTextToSpeechCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostTextToSpeech
     * @param {module:model/TTSBodyDeprecated} tTSBodyDeprecated 
     * @param {module:api/TextToSpeechApi~postTextToSpeechCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
  }, {
    key: "postTextToSpeech",
    value: function postTextToSpeech(tTSBodyDeprecated, callback) {
      var postBody = tTSBodyDeprecated;
      // verify the required parameter 'tTSBodyDeprecated' is set
      if (tTSBodyDeprecated === undefined || tTSBodyDeprecated === null) {
        throw new Error("Missing the required parameter 'tTSBodyDeprecated' when calling postTextToSpeech");
      }
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['JWT'];
      var contentTypes = ['application/json'];
      var accepts = ['text/plain'];
      var returnType = 'String';
      return this.apiClient.callApi('/v5.0/ocr/TextToSpeech/PostTextToSpeech', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
  return TextToSpeechApi;
}();
