// @asposecloud/aspose-ocr-cloud/src/model/OCRBinarizeImageBody.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
 * The OCRBinarizeImageBody model module.
 * @module model/OCRBinarizeImageBody
 * @version 5.0
 */
class OCRBinarizeImageBody {
    /**
     * Constructs a new <code>OCRBinarizeImageBody</code>.
     * @alias module:model/OCRBinarizeImageBody
     * @param image {Blob} 
     */
    constructor(image) { 
        
        OCRBinarizeImageBody.initialize(this, image);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj, image) { 
        obj['image'] = image;
    }

    /**
     * Constructs a <code>OCRBinarizeImageBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCRBinarizeImageBody} obj Optional instance to populate.
     * @return {module:model/OCRBinarizeImageBody} The populated <code>OCRBinarizeImageBody</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OCRBinarizeImageBody();

            if (data.hasOwnProperty('image')) {
                obj['image'] = ApiClient.convertToType(data['image'], 'Blob');
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>OCRBinarizeImageBody</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCRBinarizeImageBody</code>.
     */
    static validateJSON(data) {
        // check to make sure all required properties are present in the JSON string
        for (const property of OCRBinarizeImageBody.RequiredProperties) {
            if (!data[property]) {
                throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
            }
        }

        return true;
    }


}

OCRBinarizeImageBody.RequiredProperties = ["image"];

/**
 * @member {Blob} image
 */
OCRBinarizeImageBody.prototype['image'] = undefined;






export default OCRBinarizeImageBody;
