// @asposecloud/aspose-ocr-cloud/src/model/OCRRect.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
 * The OCRRect model module.
 * @module model/OCRRect
 * @version 5.0
 */
class OCRRect {
    /**
     * Constructs a new <code>OCRRect</code>.
     * Represents a rectangle: Left-Top (X1-Y1) to Right-Bottom (X2-Y2)
     * @alias module:model/OCRRect
     */
    constructor() { 
        
        OCRRect.initialize(this);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj) { 
    }

    /**
     * Constructs a <code>OCRRect</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCRRect} obj Optional instance to populate.
     * @return {module:model/OCRRect} The populated <code>OCRRect</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OCRRect();

            if (data.hasOwnProperty('topLeftX')) {
                obj['topLeftX'] = ApiClient.convertToType(data['topLeftX'], 'Number');
            }
            if (data.hasOwnProperty('topLeftY')) {
                obj['topLeftY'] = ApiClient.convertToType(data['topLeftY'], 'Number');
            }
            if (data.hasOwnProperty('bottomRightX')) {
                obj['bottomRightX'] = ApiClient.convertToType(data['bottomRightX'], 'Number');
            }
            if (data.hasOwnProperty('bottomRightY')) {
                obj['bottomRightY'] = ApiClient.convertToType(data['bottomRightY'], 'Number');
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>OCRRect</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCRRect</code>.
     */
    static validateJSON(data) {

        return true;
    }


}



/**
 * X-Coordinate of top left corner
 * @member {Number} topLeftX
 */
OCRRect.prototype['topLeftX'] = undefined;

/**
 * Y-Coordinate of top left corner
 * @member {Number} topLeftY
 */
OCRRect.prototype['topLeftY'] = undefined;

/**
 * X-Coordinate of bottom right corner
 * @member {Number} bottomRightX
 */
OCRRect.prototype['bottomRightX'] = undefined;

/**
 * Y-Coordinate of bottom right corner
 * @member {Number} bottomRightY
 */
OCRRect.prototype['bottomRightY'] = undefined;






export default OCRRect;
