// @asposecloud/aspose-ocr-cloud/src/model/OCRRecognizeAndParseInvoiceBody.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import OCRSettingsRecognizeAndParseInvoice from './OCRSettingsRecognizeAndParseInvoice';

/**
 * The OCRRecognizeAndParseInvoiceBody model module.
 * @module model/OCRRecognizeAndParseInvoiceBody
 * @version 5.0
 */
class OCRRecognizeAndParseInvoiceBody {
    /**
     * Constructs a new <code>OCRRecognizeAndParseInvoiceBody</code>.
     * @alias module:model/OCRRecognizeAndParseInvoiceBody
     * @param image {Blob} 
     * @param settings {module:model/OCRSettingsRecognizeAndParseInvoice} 
     */
    constructor(image, settings) { 
        
        OCRRecognizeAndParseInvoiceBody.initialize(this, image, settings);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj, image, settings) { 
        obj['image'] = image;
        obj['settings'] = settings;
    }

    /**
     * Constructs a <code>OCRRecognizeAndParseInvoiceBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCRRecognizeAndParseInvoiceBody} obj Optional instance to populate.
     * @return {module:model/OCRRecognizeAndParseInvoiceBody} The populated <code>OCRRecognizeAndParseInvoiceBody</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OCRRecognizeAndParseInvoiceBody();

            if (data.hasOwnProperty('image')) {
                obj['image'] = ApiClient.convertToType(data['image'], 'Blob');
            }
            if (data.hasOwnProperty('settings')) {
                obj['settings'] = OCRSettingsRecognizeAndParseInvoice.constructFromObject(data['settings']);
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>OCRRecognizeAndParseInvoiceBody</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCRRecognizeAndParseInvoiceBody</code>.
     */
    static validateJSON(data) {
        // check to make sure all required properties are present in the JSON string
        for (const property of OCRRecognizeAndParseInvoiceBody.RequiredProperties) {
            if (!data[property]) {
                throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
            }
        }
        // validate the optional field `settings`
        if (data['settings']) { // data not null
          OCRSettingsRecognizeAndParseInvoice.validateJSON(data['settings']);
        }

        return true;
    }


}

OCRRecognizeAndParseInvoiceBody.RequiredProperties = ["image", "settings"];

/**
 * @member {Blob} image
 */
OCRRecognizeAndParseInvoiceBody.prototype['image'] = undefined;

/**
 * @member {module:model/OCRSettingsRecognizeAndParseInvoice} settings
 */
OCRRecognizeAndParseInvoiceBody.prototype['settings'] = undefined;






export default OCRRecognizeAndParseInvoiceBody;
