// @asposecloud/aspose-ocr-cloud/src/model/OCRResult.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
 * The OCRResult model module.
 * @module model/OCRResult
 * @version 5.0
 */
class OCRResult {
    /**
     * Constructs a new <code>OCRResult</code>.
     * Represents information about response after OCR.
     * @alias module:model/OCRResult
     */
    constructor() { 
        
        OCRResult.initialize(this);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj) { 
    }

    /**
     * Constructs a <code>OCRResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCRResult} obj Optional instance to populate.
     * @return {module:model/OCRResult} The populated <code>OCRResult</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OCRResult();

            if (data.hasOwnProperty('type')) {
                obj['type'] = ApiClient.convertToType(data['type'], 'String');
            }
            if (data.hasOwnProperty('data')) {
                obj['data'] = ApiClient.convertToType(data['data'], 'Blob');
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>OCRResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCRResult</code>.
     */
    static validateJSON(data) {
        // ensure the json data is a string
        if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
            throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
        }

        return true;
    }


}



/**
 * File data type (extension)
 * @member {String} type
 */
OCRResult.prototype['type'] = undefined;

/**
 * File binary data
 * @member {Blob} data
 */
OCRResult.prototype['data'] = undefined;






export default OCRResult;
