// @asposecloud/aspose-ocr-cloud/src/model/OCRSettingsRecognizePdf.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import DsrConfidence from './DsrConfidence';
import DsrMode from './DsrMode';
import Language from './Language';
import OCRRegion from './OCRRegion';
import ResultType from './ResultType';
import ResultTypeTable from './ResultTypeTable';

/**
 * The OCRSettingsRecognizePdf model module.
 * @module model/OCRSettingsRecognizePdf
 * @version 5.0
 */
class OCRSettingsRecognizePdf {
    /**
     * Constructs a new <code>OCRSettingsRecognizePdf</code>.
     * OCR Process setting for Scanned multiple PDF document recognition
     * @alias module:model/OCRSettingsRecognizePdf
     */
    constructor() { 
        
        OCRSettingsRecognizePdf.initialize(this);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj) { 
    }

    /**
     * Constructs a <code>OCRSettingsRecognizePdf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCRSettingsRecognizePdf} obj Optional instance to populate.
     * @return {module:model/OCRSettingsRecognizePdf} The populated <code>OCRSettingsRecognizePdf</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new OCRSettingsRecognizePdf();

            if (data.hasOwnProperty('language')) {
                obj['language'] = Language.constructFromObject(data['language']);
            }
            if (data.hasOwnProperty('makeSkewCorrect')) {
                obj['makeSkewCorrect'] = ApiClient.convertToType(data['makeSkewCorrect'], 'Boolean');
            }
            if (data.hasOwnProperty('makeSpellCheck')) {
                obj['makeSpellCheck'] = ApiClient.convertToType(data['makeSpellCheck'], 'Boolean');
            }
            if (data.hasOwnProperty('makeContrastCorrection')) {
                obj['makeContrastCorrection'] = ApiClient.convertToType(data['makeContrastCorrection'], 'Boolean');
            }
            if (data.hasOwnProperty('dsrMode')) {
                obj['dsrMode'] = DsrMode.constructFromObject(data['dsrMode']);
            }
            if (data.hasOwnProperty('dsrConfidence')) {
                obj['dsrConfidence'] = DsrConfidence.constructFromObject(data['dsrConfidence']);
            }
            if (data.hasOwnProperty('resultType')) {
                obj['resultType'] = ResultType.constructFromObject(data['resultType']);
            }
            if (data.hasOwnProperty('Rotate')) {
                obj['Rotate'] = ApiClient.convertToType(data['Rotate'], 'Number');
            }
            if (data.hasOwnProperty('makeBinarization')) {
                obj['makeBinarization'] = ApiClient.convertToType(data['makeBinarization'], 'Boolean');
            }
            if (data.hasOwnProperty('makeUpsampling')) {
                obj['makeUpsampling'] = ApiClient.convertToType(data['makeUpsampling'], 'Boolean');
            }
            if (data.hasOwnProperty('resultTypeTable')) {
                obj['resultTypeTable'] = ResultTypeTable.constructFromObject(data['resultTypeTable']);
            }
            if (data.hasOwnProperty('regions')) {
                obj['regions'] = ApiClient.convertToType(data['regions'], [OCRRegion]);
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>OCRSettingsRecognizePdf</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCRSettingsRecognizePdf</code>.
     */
    static validateJSON(data) {
        if (data['regions']) { // data not null
            // ensure the json data is an array
            if (!Array.isArray(data['regions'])) {
                throw new Error("Expected the field `regions` to be an array in the JSON data but got " + data['regions']);
            }
            // validate the optional field `regions` (array)
            for (const item of data['regions']) {
                OCRRegion.validateJSON(item);
            };
        }

        return true;
    }


}



/**
 * @member {module:model/Language} language
 */
OCRSettingsRecognizePdf.prototype['language'] = undefined;

/**
 * Option to enable skew correction algorithm. True by default
 * @member {Boolean} makeSkewCorrect
 * @default true
 */
OCRSettingsRecognizePdf.prototype['makeSkewCorrect'] = true;

/**
 * Option to enable spell checking and correction algorithm. False by default
 * @member {Boolean} makeSpellCheck
 * @default false
 */
OCRSettingsRecognizePdf.prototype['makeSpellCheck'] = false;

/**
 * Option to enable image contrast correction algorithm. True by default
 * @member {Boolean} makeContrastCorrection
 * @default false
 */
OCRSettingsRecognizePdf.prototype['makeContrastCorrection'] = false;

/**
 * @member {module:model/DsrMode} dsrMode
 */
OCRSettingsRecognizePdf.prototype['dsrMode'] = undefined;

/**
 * @member {module:model/DsrConfidence} dsrConfidence
 */
OCRSettingsRecognizePdf.prototype['dsrConfidence'] = undefined;

/**
 * @member {module:model/ResultType} resultType
 */
OCRSettingsRecognizePdf.prototype['resultType'] = undefined;

/**
 * @member {Number} Rotate
 */
OCRSettingsRecognizePdf.prototype['Rotate'] = undefined;

/**
 * @member {Boolean} makeBinarization
 * @default true
 */
OCRSettingsRecognizePdf.prototype['makeBinarization'] = true;

/**
 * Option to enable image up-sampling algorithm to improve quality. True by default
 * @member {Boolean} makeUpsampling
 * @default false
 */
OCRSettingsRecognizePdf.prototype['makeUpsampling'] = false;

/**
 * @member {module:model/ResultTypeTable} resultTypeTable
 */
OCRSettingsRecognizePdf.prototype['resultTypeTable'] = undefined;

/**
 * @member {Array.<module:model/OCRRegion>} regions
 */
OCRSettingsRecognizePdf.prototype['regions'] = undefined;






export default OCRSettingsRecognizePdf;
