// @asposecloud/aspose-ocr-cloud/src/model/PostUpsamplingFileRequest.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
 * The PostUpsamplingFileRequest model module.
 * @module model/PostUpsamplingFileRequest
 * @version 5.0
 */
class PostUpsamplingFileRequest {
    /**
     * Constructs a new <code>PostUpsamplingFileRequest</code>.
     * @alias module:model/PostUpsamplingFileRequest
     */
    constructor() { 
        
        PostUpsamplingFileRequest.initialize(this);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj) { 
    }

    /**
     * Constructs a <code>PostUpsamplingFileRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PostUpsamplingFileRequest} obj Optional instance to populate.
     * @return {module:model/PostUpsamplingFileRequest} The populated <code>PostUpsamplingFileRequest</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new PostUpsamplingFileRequest();

            if (data.hasOwnProperty('file')) {
                obj['file'] = ApiClient.convertToType(data['file'], File);
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PostUpsamplingFileRequest</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PostUpsamplingFileRequest</code>.
     */
    static validateJSON(data) {

        return true;
    }


}



/**
 * @member {File} file
 */
PostUpsamplingFileRequest.prototype['file'] = undefined;






export default PostUpsamplingFileRequest;
