// @asposecloud/aspose-ocr-cloud/src/model/TTSBody.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import TTSSettings from './TTSSettings';

/**
 * The TTSBody model module.
 * @module model/TTSBody
 * @version 5.0
 */
class TTSBody {
    /**
     * Constructs a new <code>TTSBody</code>.
     * @alias module:model/TTSBody
     * @param text {String} 
     * @param settings {module:model/TTSSettings} 
     */
    constructor(text, settings) { 
        
        TTSBody.initialize(this, text, settings);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj, text, settings) { 
        obj['text'] = text;
        obj['settings'] = settings;
    }

    /**
     * Constructs a <code>TTSBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TTSBody} obj Optional instance to populate.
     * @return {module:model/TTSBody} The populated <code>TTSBody</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new TTSBody();

            if (data.hasOwnProperty('text')) {
                obj['text'] = ApiClient.convertToType(data['text'], 'String');
            }
            if (data.hasOwnProperty('settings')) {
                obj['settings'] = TTSSettings.constructFromObject(data['settings']);
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>TTSBody</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TTSBody</code>.
     */
    static validateJSON(data) {
        // check to make sure all required properties are present in the JSON string
        for (const property of TTSBody.RequiredProperties) {
            if (!data[property]) {
                throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
            }
        }
        // ensure the json data is a string
        if (data['text'] && !(typeof data['text'] === 'string' || data['text'] instanceof String)) {
            throw new Error("Expected the field `text` to be a primitive type in the JSON string but got " + data['text']);
        }
        // validate the optional field `settings`
        if (data['settings']) { // data not null
          TTSSettings.validateJSON(data['settings']);
        }

        return true;
    }


}

TTSBody.RequiredProperties = ["text", "settings"];

/**
 * @member {String} text
 */
TTSBody.prototype['text'] = undefined;

/**
 * @member {module:model/TTSSettings} settings
 */
TTSBody.prototype['settings'] = undefined;






export default TTSBody;
