// @asposecloud/aspose-ocr-cloud/src/model/TTSBodyDeprecated.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import LanguageTTS from './LanguageTTS';
import ResultTypeTTS from './ResultTypeTTS';

/**
 * The TTSBodyDeprecated model module.
 * @module model/TTSBodyDeprecated
 * @version 5.0
 */
class TTSBodyDeprecated {
    /**
     * Constructs a new <code>TTSBodyDeprecated</code>.
     * @alias module:model/TTSBodyDeprecated
     * @param text {String} 
     * @param language {module:model/LanguageTTS} 
     * @param resultType {module:model/ResultTypeTTS} 
     */
    constructor(text, language, resultType) { 
        
        TTSBodyDeprecated.initialize(this, text, language, resultType);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj, text, language, resultType) { 
        obj['text'] = text;
        obj['language'] = language;
        obj['resultType'] = resultType;
    }

    /**
     * Constructs a <code>TTSBodyDeprecated</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TTSBodyDeprecated} obj Optional instance to populate.
     * @return {module:model/TTSBodyDeprecated} The populated <code>TTSBodyDeprecated</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new TTSBodyDeprecated();

            if (data.hasOwnProperty('text')) {
                obj['text'] = ApiClient.convertToType(data['text'], 'String');
            }
            if (data.hasOwnProperty('language')) {
                obj['language'] = LanguageTTS.constructFromObject(data['language']);
            }
            if (data.hasOwnProperty('resultType')) {
                obj['resultType'] = ResultTypeTTS.constructFromObject(data['resultType']);
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>TTSBodyDeprecated</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TTSBodyDeprecated</code>.
     */
    static validateJSON(data) {
        // check to make sure all required properties are present in the JSON string
        for (const property of TTSBodyDeprecated.RequiredProperties) {
            if (!data[property]) {
                throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
            }
        }
        // ensure the json data is a string
        if (data['text'] && !(typeof data['text'] === 'string' || data['text'] instanceof String)) {
            throw new Error("Expected the field `text` to be a primitive type in the JSON string but got " + data['text']);
        }

        return true;
    }


}

TTSBodyDeprecated.RequiredProperties = ["text", "language", "resultType"];

/**
 * @member {String} text
 */
TTSBodyDeprecated.prototype['text'] = undefined;

/**
 * @member {module:model/LanguageTTS} language
 */
TTSBodyDeprecated.prototype['language'] = undefined;

/**
 * @member {module:model/ResultTypeTTS} resultType
 */
TTSBodyDeprecated.prototype['resultType'] = undefined;






export default TTSBodyDeprecated;
