// @asposecloud/aspose-ocr-cloud/src/model/TTSResponse.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import ResponseStatusCode from './ResponseStatusCode';
import TTSError from './TTSError';
import TTSResult from './TTSResult';
import TTSTaskStatus from './TTSTaskStatus';

/**
 * The TTSResponse model module.
 * @module model/TTSResponse
 * @version 5.0
 */
class TTSResponse {
    /**
     * Constructs a new <code>TTSResponse</code>.
     * @alias module:model/TTSResponse
     */
    constructor() { 
        
        TTSResponse.initialize(this);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj) { 
    }

    /**
     * Constructs a <code>TTSResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TTSResponse} obj Optional instance to populate.
     * @return {module:model/TTSResponse} The populated <code>TTSResponse</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new TTSResponse();

            if (data.hasOwnProperty('id')) {
                obj['id'] = ApiClient.convertToType(data['id'], 'String');
            }
            if (data.hasOwnProperty('responseStatusCode')) {
                obj['responseStatusCode'] = ResponseStatusCode.constructFromObject(data['responseStatusCode']);
            }
            if (data.hasOwnProperty('taskStatus')) {
                obj['taskStatus'] = TTSTaskStatus.constructFromObject(data['taskStatus']);
            }
            if (data.hasOwnProperty('results')) {
                obj['results'] = ApiClient.convertToType(data['results'], [TTSResult]);
            }
            if (data.hasOwnProperty('error')) {
                obj['error'] = TTSError.constructFromObject(data['error']);
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>TTSResponse</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TTSResponse</code>.
     */
    static validateJSON(data) {
        // ensure the json data is a string
        if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
            throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
        }
        if (data['results']) { // data not null
            // ensure the json data is an array
            if (!Array.isArray(data['results'])) {
                throw new Error("Expected the field `results` to be an array in the JSON data but got " + data['results']);
            }
            // validate the optional field `results` (array)
            for (const item of data['results']) {
                TTSResult.validateJSON(item);
            };
        }
        // validate the optional field `error`
        if (data['error']) { // data not null
          TTSError.validateJSON(data['error']);
        }

        return true;
    }


}



/**
 * @member {String} id
 */
TTSResponse.prototype['id'] = undefined;

/**
 * @member {module:model/ResponseStatusCode} responseStatusCode
 */
TTSResponse.prototype['responseStatusCode'] = undefined;

/**
 * @member {module:model/TTSTaskStatus} taskStatus
 */
TTSResponse.prototype['taskStatus'] = undefined;

/**
 * @member {Array.<module:model/TTSResult>} results
 */
TTSResponse.prototype['results'] = undefined;

/**
 * @member {module:model/TTSError} error
 */
TTSResponse.prototype['error'] = undefined;






export default TTSResponse;
