// @asposecloud/aspose-ocr-cloud/src/model/TTSSettings.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
import LanguageTTS from './LanguageTTS';
import ResultTypeTTS from './ResultTypeTTS';

/**
 * The TTSSettings model module.
 * @module model/TTSSettings
 * @version 5.0
 */
class TTSSettings {
    /**
     * Constructs a new <code>TTSSettings</code>.
     * @alias module:model/TTSSettings
     * @param language {module:model/LanguageTTS} 
     * @param resultType {module:model/ResultTypeTTS} 
     */
    constructor(language, resultType) { 
        
        TTSSettings.initialize(this, language, resultType);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj, language, resultType) { 
        obj['language'] = language;
        obj['resultType'] = resultType;
    }

    /**
     * Constructs a <code>TTSSettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TTSSettings} obj Optional instance to populate.
     * @return {module:model/TTSSettings} The populated <code>TTSSettings</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new TTSSettings();

            if (data.hasOwnProperty('language')) {
                obj['language'] = LanguageTTS.constructFromObject(data['language']);
            }
            if (data.hasOwnProperty('resultType')) {
                obj['resultType'] = ResultTypeTTS.constructFromObject(data['resultType']);
            }
        }
        return obj;
    }

    /**
     * Validates the JSON data with respect to <code>TTSSettings</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TTSSettings</code>.
     */
    static validateJSON(data) {
        // check to make sure all required properties are present in the JSON string
        for (const property of TTSSettings.RequiredProperties) {
            if (!data[property]) {
                throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
            }
        }

        return true;
    }


}

TTSSettings.RequiredProperties = ["language", "resultType"];

/**
 * @member {module:model/LanguageTTS} language
 */
TTSSettings.prototype['language'] = undefined;

/**
 * @member {module:model/ResultTypeTTS} resultType
 */
TTSSettings.prototype['resultType'] = undefined;






export default TTSSettings;
