// @asposecloud/aspose-ocr-cloud/src/api/BinarizeImageApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRBinarizeImageBody from '../model/OCRBinarizeImageBody';
import OCRResponse from '../model/OCRResponse';
import ProblemDetails from '../model/ProblemDetails';

/**
* BinarizeImage service.
* @module api/BinarizeImageApi
* @version 5.0
*/
export default class BinarizeImageApi {

    /**
    * Constructs a new BinarizeImageApi. 
    * @alias module:api/BinarizeImageApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelBinarizeImage operation.
     * @callback module:api/BinarizeImageApi~cancelBinarizeImageCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelBinarizeImage
     * @param {String} id 
     * @param {module:api/BinarizeImageApi~cancelBinarizeImageCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelBinarizeImage(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelBinarizeImage");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/BinarizeImage', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getBinarizeImage operation.
     * @callback module:api/BinarizeImageApi~getBinarizeImageCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetBinarizeImage
     * @param {String} id 
     * @param {module:api/BinarizeImageApi~getBinarizeImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getBinarizeImage(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getBinarizeImage");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/BinarizeImage', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postBinarizeImage operation.
     * @callback module:api/BinarizeImageApi~postBinarizeImageCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostBinarizeImage
     * @param {module:model/OCRBinarizeImageBody} oCRBinarizeImageBody 
     * @param {module:api/BinarizeImageApi~postBinarizeImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postBinarizeImage(oCRBinarizeImageBody, callback) {
      let postBody = oCRBinarizeImageBody;
      // verify the required parameter 'oCRBinarizeImageBody' is set
      if (oCRBinarizeImageBody === undefined || oCRBinarizeImageBody === null) {
        throw new Error("Missing the required parameter 'oCRBinarizeImageBody' when calling postBinarizeImage");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/BinarizeImage', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
