// @asposecloud/aspose-ocr-cloud/src/api/ConvertTextToSpeechApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import ProblemDetails from '../model/ProblemDetails';
import TTSBody from '../model/TTSBody';
import TTSResponse from '../model/TTSResponse';

/**
* ConvertTextToSpeech service.
* @module api/ConvertTextToSpeechApi
* @version 5.0
*/
export default class ConvertTextToSpeechApi {

    /**
    * Constructs a new ConvertTextToSpeechApi. 
    * @alias module:api/ConvertTextToSpeechApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelConvertTextToSpeech operation.
     * @callback module:api/ConvertTextToSpeechApi~cancelConvertTextToSpeechCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelConvertTextToSpeech
     * @param {String} id 
     * @param {module:api/ConvertTextToSpeechApi~cancelConvertTextToSpeechCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelConvertTextToSpeech(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelConvertTextToSpeech");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/ConvertTextToSpeech', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getConvertTextToSpeech operation.
     * @callback module:api/ConvertTextToSpeechApi~getConvertTextToSpeechCallback
     * @param {String} error Error message, if any.
     * @param {module:model/TTSResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetConvertTextToSpeech
     * @param {String} id 
     * @param {module:api/ConvertTextToSpeechApi~getConvertTextToSpeechCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/TTSResponse}
     */
    getConvertTextToSpeech(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getConvertTextToSpeech");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = TTSResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/ConvertTextToSpeech', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postConvertTextToSpeech operation.
     * @callback module:api/ConvertTextToSpeechApi~postConvertTextToSpeechCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostConvertTextToSpeech
     * @param {module:model/TTSBody} tTSBody 
     * @param {module:api/ConvertTextToSpeechApi~postConvertTextToSpeechCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postConvertTextToSpeech(tTSBody, callback) {
      let postBody = tTSBody;
      // verify the required parameter 'tTSBody' is set
      if (tTSBody === undefined || tTSBody === null) {
        throw new Error("Missing the required parameter 'tTSBody' when calling postConvertTextToSpeech");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/ConvertTextToSpeech', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
