// @asposecloud/aspose-ocr-cloud/src/api/ConvertTextToSpeechTrialApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import TTSBody from '../model/TTSBody';
import TTSResponse from '../model/TTSResponse';

/**
* ConvertTextToSpeechTrial service.
* @module api/ConvertTextToSpeechTrialApi
* @version 5.0
*/
export default class ConvertTextToSpeechTrialApi {

    /**
    * Constructs a new ConvertTextToSpeechTrialApi. 
    * @alias module:api/ConvertTextToSpeechTrialApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelConvertTextToSpeechTrial operation.
     * @callback module:api/ConvertTextToSpeechTrialApi~cancelConvertTextToSpeechTrialCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelConvertTextToSpeechTrial
     * @param {String} id 
     * @param {module:api/ConvertTextToSpeechTrialApi~cancelConvertTextToSpeechTrialCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelConvertTextToSpeechTrial(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelConvertTextToSpeechTrial");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/ConvertTextToSpeechTrial', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getConvertTextToSpeechTrial operation.
     * @callback module:api/ConvertTextToSpeechTrialApi~getConvertTextToSpeechTrialCallback
     * @param {String} error Error message, if any.
     * @param {module:model/TTSResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetConvertTextToSpeechTrial
     * @param {String} id 
     * @param {module:api/ConvertTextToSpeechTrialApi~getConvertTextToSpeechTrialCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/TTSResponse}
     */
    getConvertTextToSpeechTrial(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getConvertTextToSpeechTrial");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = TTSResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/ConvertTextToSpeechTrial', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postConvertTextToSpeechTrial operation.
     * @callback module:api/ConvertTextToSpeechTrialApi~postConvertTextToSpeechTrialCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostConvertTextToSpeechTrial
     * @param {module:model/TTSBody} tTSBody 
     * @param {module:api/ConvertTextToSpeechTrialApi~postConvertTextToSpeechTrialCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postConvertTextToSpeechTrial(tTSBody, callback) {
      let postBody = tTSBody;
      // verify the required parameter 'tTSBody' is set
      if (tTSBody === undefined || tTSBody === null) {
        throw new Error("Missing the required parameter 'tTSBody' when calling postConvertTextToSpeechTrial");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/ConvertTextToSpeechTrial', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
