// @asposecloud/aspose-ocr-cloud/src/api/DetectRegionsApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRDetectRegionsBody from '../model/OCRDetectRegionsBody';
import OCRResponse from '../model/OCRResponse';
import ProblemDetails from '../model/ProblemDetails';

/**
* DetectRegions service.
* @module api/DetectRegionsApi
* @version 5.0
*/
export default class DetectRegionsApi {

    /**
    * Constructs a new DetectRegionsApi. 
    * @alias module:api/DetectRegionsApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelDetectRegions operation.
     * @callback module:api/DetectRegionsApi~cancelDetectRegionsCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelDetectRegions
     * @param {String} id 
     * @param {module:api/DetectRegionsApi~cancelDetectRegionsCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelDetectRegions(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelDetectRegions");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/DetectRegions', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getDetectRegions operation.
     * @callback module:api/DetectRegionsApi~getDetectRegionsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetDetectRegions
     * @param {String} id 
     * @param {module:api/DetectRegionsApi~getDetectRegionsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getDetectRegions(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getDetectRegions");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/DetectRegions', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postDetectRegions operation.
     * @callback module:api/DetectRegionsApi~postDetectRegionsCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostDetectRegions
     * @param {module:model/OCRDetectRegionsBody} oCRDetectRegionsBody 
     * @param {module:api/DetectRegionsApi~postDetectRegionsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postDetectRegions(oCRDetectRegionsBody, callback) {
      let postBody = oCRDetectRegionsBody;
      // verify the required parameter 'oCRDetectRegionsBody' is set
      if (oCRDetectRegionsBody === undefined || oCRDetectRegionsBody === null) {
        throw new Error("Missing the required parameter 'oCRDetectRegionsBody' when calling postDetectRegions");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/DetectRegions', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
