// @asposecloud/aspose-ocr-cloud/src/api/DewarpImageApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRDewarpImageBody from '../model/OCRDewarpImageBody';
import OCRResponse from '../model/OCRResponse';
import ProblemDetails from '../model/ProblemDetails';

/**
* DewarpImage service.
* @module api/DewarpImageApi
* @version 5.0
*/
export default class DewarpImageApi {

    /**
    * Constructs a new DewarpImageApi. 
    * @alias module:api/DewarpImageApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelDewarpImage operation.
     * @callback module:api/DewarpImageApi~cancelDewarpImageCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelDewarpImage
     * @param {String} id 
     * @param {module:api/DewarpImageApi~cancelDewarpImageCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelDewarpImage(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelDewarpImage");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/DewarpImage', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getDewarpImage operation.
     * @callback module:api/DewarpImageApi~getDewarpImageCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetDewarpImage
     * @param {String} id 
     * @param {module:api/DewarpImageApi~getDewarpImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getDewarpImage(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getDewarpImage");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/DewarpImage', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postDewarpImage operation.
     * @callback module:api/DewarpImageApi~postDewarpImageCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostDewarpImage
     * @param {module:model/OCRDewarpImageBody} oCRDewarpImageBody 
     * @param {module:api/DewarpImageApi~postDewarpImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postDewarpImage(oCRDewarpImageBody, callback) {
      let postBody = oCRDewarpImageBody;
      // verify the required parameter 'oCRDewarpImageBody' is set
      if (oCRDewarpImageBody === undefined || oCRDewarpImageBody === null) {
        throw new Error("Missing the required parameter 'oCRDewarpImageBody' when calling postDewarpImage");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/DewarpImage', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
