// @asposecloud/aspose-ocr-cloud/src/api/DjVu2PDFApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRDjVu2PDFBody from '../model/OCRDjVu2PDFBody';
import OCRResponse from '../model/OCRResponse';
import ProblemDetails from '../model/ProblemDetails';

/**
* DjVu2PDF service.
* @module api/DjVu2PDFApi
* @version 5.0
*/
export default class DjVu2PDFApi {

    /**
    * Constructs a new DjVu2PDFApi. 
    * @alias module:api/DjVu2PDFApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelDjVu2PDF operation.
     * @callback module:api/DjVu2PDFApi~cancelDjVu2PDFCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelDjVu2PDF
     * @param {String} id 
     * @param {module:api/DjVu2PDFApi~cancelDjVu2PDFCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelDjVu2PDF(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelDjVu2PDF");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/DjVu2PDF', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getDjVu2PDF operation.
     * @callback module:api/DjVu2PDFApi~getDjVu2PDFCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetDjVu2PDF
     * @param {String} id 
     * @param {module:api/DjVu2PDFApi~getDjVu2PDFCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getDjVu2PDF(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getDjVu2PDF");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/DjVu2PDF', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postDjVu2PDF operation.
     * @callback module:api/DjVu2PDFApi~postDjVu2PDFCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostDjVu2PDF
     * @param {module:model/OCRDjVu2PDFBody} oCRDjVu2PDFBody 
     * @param {module:api/DjVu2PDFApi~postDjVu2PDFCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postDjVu2PDF(oCRDjVu2PDFBody, callback) {
      let postBody = oCRDjVu2PDFBody;
      // verify the required parameter 'oCRDjVu2PDFBody' is set
      if (oCRDjVu2PDFBody === undefined || oCRDjVu2PDFBody === null) {
        throw new Error("Missing the required parameter 'oCRDjVu2PDFBody' when calling postDjVu2PDF");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/DjVu2PDF', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
