// @asposecloud/aspose-ocr-cloud/src/api/IdentifyFontApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRRecognizeFontBody from '../model/OCRRecognizeFontBody';
import OCRResponse from '../model/OCRResponse';

/**
* IdentifyFont service.
* @module api/IdentifyFontApi
* @version 5.0
*/
export default class IdentifyFontApi {

    /**
    * Constructs a new IdentifyFontApi. 
    * @alias module:api/IdentifyFontApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelIdentifyFont operation.
     * @callback module:api/IdentifyFontApi~cancelIdentifyFontCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelIdentifyFont
     * @param {String} id 
     * @param {module:api/IdentifyFontApi~cancelIdentifyFontCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelIdentifyFont(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelIdentifyFont");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/IdentifyFont', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getIdentifyFont operation.
     * @callback module:api/IdentifyFontApi~getIdentifyFontCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetIdentifyFont
     * @param {String} id 
     * @param {module:api/IdentifyFontApi~getIdentifyFontCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getIdentifyFont(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getIdentifyFont");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/IdentifyFont', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postIdentifyFont operation.
     * @callback module:api/IdentifyFontApi~postIdentifyFontCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostIdentifyFont
     * @param {module:model/OCRRecognizeFontBody} oCRRecognizeFontBody 
     * @param {module:api/IdentifyFontApi~postIdentifyFontCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postIdentifyFont(oCRRecognizeFontBody, callback) {
      let postBody = oCRRecognizeFontBody;
      // verify the required parameter 'oCRRecognizeFontBody' is set
      if (oCRRecognizeFontBody === undefined || oCRRecognizeFontBody === null) {
        throw new Error("Missing the required parameter 'oCRRecognizeFontBody' when calling postIdentifyFont");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/IdentifyFont', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
