// @asposecloud/aspose-ocr-cloud/src/api/RecognizeAndParseInvoiceApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRRecognizeAndParseInvoiceBody from '../model/OCRRecognizeAndParseInvoiceBody';
import OCRResponse from '../model/OCRResponse';
import ProblemDetails from '../model/ProblemDetails';

/**
* RecognizeAndParseInvoice service.
* @module api/RecognizeAndParseInvoiceApi
* @version 5.0
*/
export default class RecognizeAndParseInvoiceApi {

    /**
    * Constructs a new RecognizeAndParseInvoiceApi. 
    * @alias module:api/RecognizeAndParseInvoiceApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelRecognizeAndParseInvoice operation.
     * @callback module:api/RecognizeAndParseInvoiceApi~cancelRecognizeAndParseInvoiceCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelRecognizeAndParseInvoice
     * @param {String} id 
     * @param {module:api/RecognizeAndParseInvoiceApi~cancelRecognizeAndParseInvoiceCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelRecognizeAndParseInvoice(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelRecognizeAndParseInvoice");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeAndParseInvoice', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getRecognizeAndParseInvoice operation.
     * @callback module:api/RecognizeAndParseInvoiceApi~getRecognizeAndParseInvoiceCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetRecognizeAndParseInvoice
     * @param {String} id 
     * @param {module:api/RecognizeAndParseInvoiceApi~getRecognizeAndParseInvoiceCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getRecognizeAndParseInvoice(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getRecognizeAndParseInvoice");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeAndParseInvoice', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postRecognizeAndParseInvoice operation.
     * @callback module:api/RecognizeAndParseInvoiceApi~postRecognizeAndParseInvoiceCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostRecognizeAndParseInvoice
     * @param {module:model/OCRRecognizeAndParseInvoiceBody} oCRRecognizeAndParseInvoiceBody 
     * @param {module:api/RecognizeAndParseInvoiceApi~postRecognizeAndParseInvoiceCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postRecognizeAndParseInvoice(oCRRecognizeAndParseInvoiceBody, callback) {
      let postBody = oCRRecognizeAndParseInvoiceBody;
      // verify the required parameter 'oCRRecognizeAndParseInvoiceBody' is set
      if (oCRRecognizeAndParseInvoiceBody === undefined || oCRRecognizeAndParseInvoiceBody === null) {
        throw new Error("Missing the required parameter 'oCRRecognizeAndParseInvoiceBody' when calling postRecognizeAndParseInvoice");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeAndParseInvoice', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
