// @asposecloud/aspose-ocr-cloud/src/api/RecognizeReceiptApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRRecognizeReceiptBody from '../model/OCRRecognizeReceiptBody';
import OCRResponse from '../model/OCRResponse';
import ProblemDetails from '../model/ProblemDetails';

/**
* RecognizeReceipt service.
* @module api/RecognizeReceiptApi
* @version 5.0
*/
export default class RecognizeReceiptApi {

    /**
    * Constructs a new RecognizeReceiptApi. 
    * @alias module:api/RecognizeReceiptApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelRecognizeReceipt operation.
     * @callback module:api/RecognizeReceiptApi~cancelRecognizeReceiptCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelRecognizeReceipt
     * @param {String} id 
     * @param {module:api/RecognizeReceiptApi~cancelRecognizeReceiptCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelRecognizeReceipt(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelRecognizeReceipt");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeReceipt', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getRecognizeReceipt operation.
     * @callback module:api/RecognizeReceiptApi~getRecognizeReceiptCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetRecognizeReceipt
     * @param {String} id 
     * @param {module:api/RecognizeReceiptApi~getRecognizeReceiptCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getRecognizeReceipt(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getRecognizeReceipt");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeReceipt', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postRecognizeReceipt operation.
     * @callback module:api/RecognizeReceiptApi~postRecognizeReceiptCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostRecognizeReceipt
     * @param {module:model/OCRRecognizeReceiptBody} oCRRecognizeReceiptBody 
     * @param {module:api/RecognizeReceiptApi~postRecognizeReceiptCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postRecognizeReceipt(oCRRecognizeReceiptBody, callback) {
      let postBody = oCRRecognizeReceiptBody;
      // verify the required parameter 'oCRRecognizeReceiptBody' is set
      if (oCRRecognizeReceiptBody === undefined || oCRRecognizeReceiptBody === null) {
        throw new Error("Missing the required parameter 'oCRRecognizeReceiptBody' when calling postRecognizeReceipt");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeReceipt', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
