// @asposecloud/aspose-ocr-cloud/src/api/RecognizeImageTrialApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRRecognizeImageBody from '../model/OCRRecognizeImageBody';
import OCRResponse from '../model/OCRResponse';

/**
* RecognizeImageTrial service.
* @module api/RecognizeImageTrialApi
* @version 5.0
*/
export default class RecognizeImageTrialApi {

    /**
    * Constructs a new RecognizeImageTrialApi. 
    * @alias module:api/RecognizeImageTrialApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelRecognizeImageTrial operation.
     * @callback module:api/RecognizeImageTrialApi~cancelRecognizeImageTrialCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelRecognizeImageTrial
     * @param {String} id 
     * @param {module:api/RecognizeImageTrialApi~cancelRecognizeImageTrialCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelRecognizeImageTrial(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelRecognizeImageTrial");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeImageTrial', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getRecognizeImageTrial operation.
     * @callback module:api/RecognizeImageTrialApi~getRecognizeImageTrialCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetRecognizeImageTrial
     * @param {String} id 
     * @param {module:api/RecognizeImageTrialApi~getRecognizeImageTrialCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getRecognizeImageTrial(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getRecognizeImageTrial");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeImageTrial', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postRecognizeImageTrial operation.
     * @callback module:api/RecognizeImageTrialApi~postRecognizeImageTrialCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostRecognizeImageTrial
     * @param {module:model/OCRRecognizeImageBody} oCRRecognizeImageBody 
     * @param {module:api/RecognizeImageTrialApi~postRecognizeImageTrialCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postRecognizeImageTrial(oCRRecognizeImageBody, callback) {
      let postBody = oCRRecognizeImageBody;
      // verify the required parameter 'oCRRecognizeImageBody' is set
      if (oCRRecognizeImageBody === undefined || oCRRecognizeImageBody === null) {
        throw new Error("Missing the required parameter 'oCRRecognizeImageBody' when calling postRecognizeImageTrial");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeImageTrial', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
