// @asposecloud/aspose-ocr-cloud/src/api/RecognizeLabelApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRRecognizeLabelBody from '../model/OCRRecognizeLabelBody';
import OCRResponse from '../model/OCRResponse';
import ProblemDetails from '../model/ProblemDetails';

/**
* RecognizeLabel service.
* @module api/RecognizeLabelApi
* @version 5.0
*/
export default class RecognizeLabelApi {

    /**
    * Constructs a new RecognizeLabelApi. 
    * @alias module:api/RecognizeLabelApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelRecognizeLabel operation.
     * @callback module:api/RecognizeLabelApi~cancelRecognizeLabelCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelRecognizeLabel
     * @param {String} id 
     * @param {module:api/RecognizeLabelApi~cancelRecognizeLabelCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelRecognizeLabel(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelRecognizeLabel");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeLabel', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getRecognizeLabel operation.
     * @callback module:api/RecognizeLabelApi~getRecognizeLabelCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetRecognizeLabel
     * @param {String} id 
     * @param {module:api/RecognizeLabelApi~getRecognizeLabelCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getRecognizeLabel(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getRecognizeLabel");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeLabel', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postRecognizeLabel operation.
     * @callback module:api/RecognizeLabelApi~postRecognizeLabelCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostRecognizeLabel
     * @param {module:model/OCRRecognizeLabelBody} oCRRecognizeLabelBody 
     * @param {module:api/RecognizeLabelApi~postRecognizeLabelCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postRecognizeLabel(oCRRecognizeLabelBody, callback) {
      let postBody = oCRRecognizeLabelBody;
      // verify the required parameter 'oCRRecognizeLabelBody' is set
      if (oCRRecognizeLabelBody === undefined || oCRRecognizeLabelBody === null) {
        throw new Error("Missing the required parameter 'oCRRecognizeLabelBody' when calling postRecognizeLabel");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/RecognizeLabel', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
