// @asposecloud/aspose-ocr-cloud/src/api/TextToSpeechApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import ProblemDetails from '../model/ProblemDetails';
import TTSBodyDeprecated from '../model/TTSBodyDeprecated';
import TTSResponse from '../model/TTSResponse';

/**
* TextToSpeech service.
* @module api/TextToSpeechApi
* @version 5.0
*/
export default class TextToSpeechApi {

    /**
    * Constructs a new TextToSpeechApi. 
    * @alias module:api/TextToSpeechApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the getTextToSpeechResult operation.
     * @callback module:api/TextToSpeechApi~getTextToSpeechResultCallback
     * @param {String} error Error message, if any.
     * @param {module:model/TTSResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetTextToSpeechResult
     * @param {String} id 
     * @param {module:api/TextToSpeechApi~getTextToSpeechResultCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/TTSResponse}
     */
    getTextToSpeechResult(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getTextToSpeechResult");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = TTSResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/TextToSpeech/GetTextToSpeechResult', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getTextToSpeechResultFile operation.
     * @callback module:api/TextToSpeechApi~getTextToSpeechResultFileCallback
     * @param {String} error Error message, if any.
     * @param {Object} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetTextToSpeechResultFile
     * @param {String} id 
     * @param {module:api/TextToSpeechApi~getTextToSpeechResultFileCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Object}
     */
    getTextToSpeechResultFile(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getTextToSpeechResultFile");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/octet-stream'];
      let returnType = Object;
      return this.apiClient.callApi(
        '/v5.0/ocr/TextToSpeech/GetTextToSpeechResultFile', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postTextToSpeech operation.
     * @callback module:api/TextToSpeechApi~postTextToSpeechCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostTextToSpeech
     * @param {module:model/TTSBodyDeprecated} tTSBodyDeprecated 
     * @param {module:api/TextToSpeechApi~postTextToSpeechCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postTextToSpeech(tTSBodyDeprecated, callback) {
      let postBody = tTSBodyDeprecated;
      // verify the required parameter 'tTSBodyDeprecated' is set
      if (tTSBodyDeprecated === undefined || tTSBodyDeprecated === null) {
        throw new Error("Missing the required parameter 'tTSBodyDeprecated' when calling postTextToSpeech");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/TextToSpeech/PostTextToSpeech', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
