// @asposecloud/aspose-ocr-cloud/src/api/UpscaleImageApi.js
/**
 * Aspose OCR Cloud 5.0 API
 * Aspose OCR Cloud 5.0 API
 *
 * The version of the OpenAPI document: 5.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import OCRResponse from '../model/OCRResponse';
import OCRUpscaleImageBody from '../model/OCRUpscaleImageBody';
import ProblemDetails from '../model/ProblemDetails';

/**
* UpscaleImage service.
* @module api/UpscaleImageApi
* @version 5.0
*/
export default class UpscaleImageApi {

    /**
    * Constructs a new UpscaleImageApi. 
    * @alias module:api/UpscaleImageApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }


    /**
     * Callback function to receive the result of the cancelUpscaleImage operation.
     * @callback module:api/UpscaleImageApi~cancelUpscaleImageCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * CancelUpscaleImage
     * @param {String} id 
     * @param {module:api/UpscaleImageApi~cancelUpscaleImageCallback} callback The callback function, accepting three arguments: error, data, response
     */
    cancelUpscaleImage(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling cancelUpscaleImage");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = [];
      let returnType = null;
      return this.apiClient.callApi(
        '/v5.0/ocr/UpscaleImage', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the getUpscaleImage operation.
     * @callback module:api/UpscaleImageApi~getUpscaleImageCallback
     * @param {String} error Error message, if any.
     * @param {module:model/OCRResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * GetUpscaleImage
     * @param {String} id 
     * @param {module:api/UpscaleImageApi~getUpscaleImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/OCRResponse}
     */
    getUpscaleImage(id, callback) {
      let postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getUpscaleImage");
      }

      let pathParams = {
      };
      let queryParams = {
        'id': id
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = OCRResponse;
      return this.apiClient.callApi(
        '/v5.0/ocr/UpscaleImage', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }

    /**
     * Callback function to receive the result of the postUpscaleImage operation.
     * @callback module:api/UpscaleImageApi~postUpscaleImageCallback
     * @param {String} error Error message, if any.
     * @param {String} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * PostUpscaleImage
     * @param {module:model/OCRUpscaleImageBody} oCRUpscaleImageBody 
     * @param {module:api/UpscaleImageApi~postUpscaleImageCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link String}
     */
    postUpscaleImage(oCRUpscaleImageBody, callback) {
      let postBody = oCRUpscaleImageBody;
      // verify the required parameter 'oCRUpscaleImageBody' is set
      if (oCRUpscaleImageBody === undefined || oCRUpscaleImageBody === null) {
        throw new Error("Missing the required parameter 'oCRUpscaleImageBody' when calling postUpscaleImage");
      }

      let pathParams = {
      };
      let queryParams = {
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['JWT'];
      let contentTypes = ['application/json'];
      let accepts = ['text/plain'];
      let returnType = 'String';
      return this.apiClient.callApi(
        '/v5.0/ocr/UpscaleImage', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null, callback
      );
    }


}
