// @asposecloud/aspose-ocr-cloud/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "BinarizeImageApi", {
  enumerable: true,
  get: function get() {
    return _BinarizeImageApi["default"];
  }
});
Object.defineProperty(exports, "ConvertTextToSpeechApi", {
  enumerable: true,
  get: function get() {
    return _ConvertTextToSpeechApi["default"];
  }
});
Object.defineProperty(exports, "ConvertTextToSpeechTrialApi", {
  enumerable: true,
  get: function get() {
    return _ConvertTextToSpeechTrialApi["default"];
  }
});
Object.defineProperty(exports, "DeskewImageApi", {
  enumerable: true,
  get: function get() {
    return _DeskewImageApi["default"];
  }
});
Object.defineProperty(exports, "DetectRegionsApi", {
  enumerable: true,
  get: function get() {
    return _DetectRegionsApi["default"];
  }
});
Object.defineProperty(exports, "DewarpImageApi", {
  enumerable: true,
  get: function get() {
    return _DewarpImageApi["default"];
  }
});
Object.defineProperty(exports, "DjVu2PDFApi", {
  enumerable: true,
  get: function get() {
    return _DjVu2PDFApi["default"];
  }
});
Object.defineProperty(exports, "DsrConfidence", {
  enumerable: true,
  get: function get() {
    return _DsrConfidence["default"];
  }
});
Object.defineProperty(exports, "DsrMode", {
  enumerable: true,
  get: function get() {
    return _DsrMode["default"];
  }
});
Object.defineProperty(exports, "IdentifyFontApi", {
  enumerable: true,
  get: function get() {
    return _IdentifyFontApi["default"];
  }
});
Object.defineProperty(exports, "ImageProcessingApi", {
  enumerable: true,
  get: function get() {
    return _ImageProcessingApi["default"];
  }
});
Object.defineProperty(exports, "Language", {
  enumerable: true,
  get: function get() {
    return _Language["default"];
  }
});
Object.defineProperty(exports, "LanguageTTS", {
  enumerable: true,
  get: function get() {
    return _LanguageTTS["default"];
  }
});
Object.defineProperty(exports, "OCRBinarizeImageBody", {
  enumerable: true,
  get: function get() {
    return _OCRBinarizeImageBody["default"];
  }
});
Object.defineProperty(exports, "OCRDeskewImageBody", {
  enumerable: true,
  get: function get() {
    return _OCRDeskewImageBody["default"];
  }
});
Object.defineProperty(exports, "OCRDetectRegionsBody", {
  enumerable: true,
  get: function get() {
    return _OCRDetectRegionsBody["default"];
  }
});
Object.defineProperty(exports, "OCRDewarpImageBody", {
  enumerable: true,
  get: function get() {
    return _OCRDewarpImageBody["default"];
  }
});
Object.defineProperty(exports, "OCRDjVu2PDFBody", {
  enumerable: true,
  get: function get() {
    return _OCRDjVu2PDFBody["default"];
  }
});
Object.defineProperty(exports, "OCRError", {
  enumerable: true,
  get: function get() {
    return _OCRError["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizeAndParseInvoiceBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizeAndParseInvoiceBody["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizeFontBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizeFontBody["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizeImageBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizeImageBody["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizeLabelBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizeLabelBody["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizePdfBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizePdfBody["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizeReceiptBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizeReceiptBody["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizeRegionsBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizeRegionsBody["default"];
  }
});
Object.defineProperty(exports, "OCRRecognizeTableBody", {
  enumerable: true,
  get: function get() {
    return _OCRRecognizeTableBody["default"];
  }
});
Object.defineProperty(exports, "OCRRect", {
  enumerable: true,
  get: function get() {
    return _OCRRect["default"];
  }
});
Object.defineProperty(exports, "OCRRegion", {
  enumerable: true,
  get: function get() {
    return _OCRRegion["default"];
  }
});
Object.defineProperty(exports, "OCRResponse", {
  enumerable: true,
  get: function get() {
    return _OCRResponse["default"];
  }
});
Object.defineProperty(exports, "OCRResult", {
  enumerable: true,
  get: function get() {
    return _OCRResult["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsDetectRegions", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsDetectRegions["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsDjVu2PDF", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsDjVu2PDF["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizeAndParseInvoice", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizeAndParseInvoice["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizeFont", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizeFont["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizeImage", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizeImage["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizeLabel", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizeLabel["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizePdf", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizePdf["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizeReceipt", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizeReceipt["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizeRegions", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizeRegions["default"];
  }
});
Object.defineProperty(exports, "OCRSettingsRecognizeTable", {
  enumerable: true,
  get: function get() {
    return _OCRSettingsRecognizeTable["default"];
  }
});
Object.defineProperty(exports, "OCRTaskStatus", {
  enumerable: true,
  get: function get() {
    return _OCRTaskStatus["default"];
  }
});
Object.defineProperty(exports, "OCRUpscaleImageBody", {
  enumerable: true,
  get: function get() {
    return _OCRUpscaleImageBody["default"];
  }
});
Object.defineProperty(exports, "PostUpsamplingFileRequest", {
  enumerable: true,
  get: function get() {
    return _PostUpsamplingFileRequest["default"];
  }
});
Object.defineProperty(exports, "ProblemDetails", {
  enumerable: true,
  get: function get() {
    return _ProblemDetails["default"];
  }
});
Object.defineProperty(exports, "RecognizeAndParseInvoiceApi", {
  enumerable: true,
  get: function get() {
    return _RecognizeAndParseInvoiceApi["default"];
  }
});
Object.defineProperty(exports, "RecognizeImageApi", {
  enumerable: true,
  get: function get() {
    return _RecognizeImageApi["default"];
  }
});
Object.defineProperty(exports, "RecognizeImageTrialApi", {
  enumerable: true,
  get: function get() {
    return _RecognizeImageTrialApi["default"];
  }
});
Object.defineProperty(exports, "RecognizeLabelApi", {
  enumerable: true,
  get: function get() {
    return _RecognizeLabelApi["default"];
  }
});
Object.defineProperty(exports, "RecognizePdfApi", {
  enumerable: true,
  get: function get() {
    return _RecognizePdfApi["default"];
  }
});
Object.defineProperty(exports, "RecognizeReceiptApi", {
  enumerable: true,
  get: function get() {
    return _RecognizeReceiptApi["default"];
  }
});
Object.defineProperty(exports, "RecognizeRegionsApi", {
  enumerable: true,
  get: function get() {
    return _RecognizeRegionsApi["default"];
  }
});
Object.defineProperty(exports, "RecognizeTableApi", {
  enumerable: true,
  get: function get() {
    return _RecognizeTableApi["default"];
  }
});
Object.defineProperty(exports, "ResponseStatusCode", {
  enumerable: true,
  get: function get() {
    return _ResponseStatusCode["default"];
  }
});
Object.defineProperty(exports, "ResultType", {
  enumerable: true,
  get: function get() {
    return _ResultType["default"];
  }
});
Object.defineProperty(exports, "ResultTypeTTS", {
  enumerable: true,
  get: function get() {
    return _ResultTypeTTS["default"];
  }
});
Object.defineProperty(exports, "ResultTypeTable", {
  enumerable: true,
  get: function get() {
    return _ResultTypeTable["default"];
  }
});
Object.defineProperty(exports, "TTSBody", {
  enumerable: true,
  get: function get() {
    return _TTSBody["default"];
  }
});
Object.defineProperty(exports, "TTSBodyDeprecated", {
  enumerable: true,
  get: function get() {
    return _TTSBodyDeprecated["default"];
  }
});
Object.defineProperty(exports, "TTSError", {
  enumerable: true,
  get: function get() {
    return _TTSError["default"];
  }
});
Object.defineProperty(exports, "TTSResponse", {
  enumerable: true,
  get: function get() {
    return _TTSResponse["default"];
  }
});
Object.defineProperty(exports, "TTSResult", {
  enumerable: true,
  get: function get() {
    return _TTSResult["default"];
  }
});
Object.defineProperty(exports, "TTSSettings", {
  enumerable: true,
  get: function get() {
    return _TTSSettings["default"];
  }
});
Object.defineProperty(exports, "TTSTaskStatus", {
  enumerable: true,
  get: function get() {
    return _TTSTaskStatus["default"];
  }
});
Object.defineProperty(exports, "TextToSpeechApi", {
  enumerable: true,
  get: function get() {
    return _TextToSpeechApi["default"];
  }
});
Object.defineProperty(exports, "UpscaleImageApi", {
  enumerable: true,
  get: function get() {
    return _UpscaleImageApi["default"];
  }
});
Object.defineProperty(exports, "UtilitiesApi", {
  enumerable: true,
  get: function get() {
    return _UtilitiesApi["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _DsrConfidence = _interopRequireDefault(require("./model/DsrConfidence"));
var _DsrMode = _interopRequireDefault(require("./model/DsrMode"));
var _Language = _interopRequireDefault(require("./model/Language"));
var _LanguageTTS = _interopRequireDefault(require("./model/LanguageTTS"));
var _OCRBinarizeImageBody = _interopRequireDefault(require("./model/OCRBinarizeImageBody"));
var _OCRDeskewImageBody = _interopRequireDefault(require("./model/OCRDeskewImageBody"));
var _OCRDetectRegionsBody = _interopRequireDefault(require("./model/OCRDetectRegionsBody"));
var _OCRDewarpImageBody = _interopRequireDefault(require("./model/OCRDewarpImageBody"));
var _OCRDjVu2PDFBody = _interopRequireDefault(require("./model/OCRDjVu2PDFBody"));
var _OCRError = _interopRequireDefault(require("./model/OCRError"));
var _OCRRecognizeAndParseInvoiceBody = _interopRequireDefault(require("./model/OCRRecognizeAndParseInvoiceBody"));
var _OCRRecognizeFontBody = _interopRequireDefault(require("./model/OCRRecognizeFontBody"));
var _OCRRecognizeImageBody = _interopRequireDefault(require("./model/OCRRecognizeImageBody"));
var _OCRRecognizeLabelBody = _interopRequireDefault(require("./model/OCRRecognizeLabelBody"));
var _OCRRecognizePdfBody = _interopRequireDefault(require("./model/OCRRecognizePdfBody"));
var _OCRRecognizeReceiptBody = _interopRequireDefault(require("./model/OCRRecognizeReceiptBody"));
var _OCRRecognizeRegionsBody = _interopRequireDefault(require("./model/OCRRecognizeRegionsBody"));
var _OCRRecognizeTableBody = _interopRequireDefault(require("./model/OCRRecognizeTableBody"));
var _OCRRect = _interopRequireDefault(require("./model/OCRRect"));
var _OCRRegion = _interopRequireDefault(require("./model/OCRRegion"));
var _OCRResponse = _interopRequireDefault(require("./model/OCRResponse"));
var _OCRResult = _interopRequireDefault(require("./model/OCRResult"));
var _OCRSettingsDetectRegions = _interopRequireDefault(require("./model/OCRSettingsDetectRegions"));
var _OCRSettingsDjVu2PDF = _interopRequireDefault(require("./model/OCRSettingsDjVu2PDF"));
var _OCRSettingsRecognizeAndParseInvoice = _interopRequireDefault(require("./model/OCRSettingsRecognizeAndParseInvoice"));
var _OCRSettingsRecognizeFont = _interopRequireDefault(require("./model/OCRSettingsRecognizeFont"));
var _OCRSettingsRecognizeImage = _interopRequireDefault(require("./model/OCRSettingsRecognizeImage"));
var _OCRSettingsRecognizeLabel = _interopRequireDefault(require("./model/OCRSettingsRecognizeLabel"));
var _OCRSettingsRecognizePdf = _interopRequireDefault(require("./model/OCRSettingsRecognizePdf"));
var _OCRSettingsRecognizeReceipt = _interopRequireDefault(require("./model/OCRSettingsRecognizeReceipt"));
var _OCRSettingsRecognizeRegions = _interopRequireDefault(require("./model/OCRSettingsRecognizeRegions"));
var _OCRSettingsRecognizeTable = _interopRequireDefault(require("./model/OCRSettingsRecognizeTable"));
var _OCRTaskStatus = _interopRequireDefault(require("./model/OCRTaskStatus"));
var _OCRUpscaleImageBody = _interopRequireDefault(require("./model/OCRUpscaleImageBody"));
var _PostUpsamplingFileRequest = _interopRequireDefault(require("./model/PostUpsamplingFileRequest"));
var _ProblemDetails = _interopRequireDefault(require("./model/ProblemDetails"));
var _ResponseStatusCode = _interopRequireDefault(require("./model/ResponseStatusCode"));
var _ResultType = _interopRequireDefault(require("./model/ResultType"));
var _ResultTypeTTS = _interopRequireDefault(require("./model/ResultTypeTTS"));
var _ResultTypeTable = _interopRequireDefault(require("./model/ResultTypeTable"));
var _TTSBody = _interopRequireDefault(require("./model/TTSBody"));
var _TTSBodyDeprecated = _interopRequireDefault(require("./model/TTSBodyDeprecated"));
var _TTSError = _interopRequireDefault(require("./model/TTSError"));
var _TTSResponse = _interopRequireDefault(require("./model/TTSResponse"));
var _TTSResult = _interopRequireDefault(require("./model/TTSResult"));
var _TTSSettings = _interopRequireDefault(require("./model/TTSSettings"));
var _TTSTaskStatus = _interopRequireDefault(require("./model/TTSTaskStatus"));
var _BinarizeImageApi = _interopRequireDefault(require("./api/BinarizeImageApi"));
var _ConvertTextToSpeechApi = _interopRequireDefault(require("./api/ConvertTextToSpeechApi"));
var _ConvertTextToSpeechTrialApi = _interopRequireDefault(require("./api/ConvertTextToSpeechTrialApi"));
var _DeskewImageApi = _interopRequireDefault(require("./api/DeskewImageApi"));
var _DetectRegionsApi = _interopRequireDefault(require("./api/DetectRegionsApi"));
var _DewarpImageApi = _interopRequireDefault(require("./api/DewarpImageApi"));
var _DjVu2PDFApi = _interopRequireDefault(require("./api/DjVu2PDFApi"));
var _IdentifyFontApi = _interopRequireDefault(require("./api/IdentifyFontApi"));
var _ImageProcessingApi = _interopRequireDefault(require("./api/ImageProcessingApi"));
var _RecognizeAndParseInvoiceApi = _interopRequireDefault(require("./api/RecognizeAndParseInvoiceApi"));
var _RecognizeImageApi = _interopRequireDefault(require("./api/RecognizeImageApi"));
var _RecognizeImageTrialApi = _interopRequireDefault(require("./api/RecognizeImageTrialApi"));
var _RecognizeLabelApi = _interopRequireDefault(require("./api/RecognizeLabelApi"));
var _RecognizePdfApi = _interopRequireDefault(require("./api/RecognizePdfApi"));
var _RecognizeReceiptApi = _interopRequireDefault(require("./api/RecognizeReceiptApi"));
var _RecognizeRegionsApi = _interopRequireDefault(require("./api/RecognizeRegionsApi"));
var _RecognizeTableApi = _interopRequireDefault(require("./api/RecognizeTableApi"));
var _TextToSpeechApi = _interopRequireDefault(require("./api/TextToSpeechApi"));
var _UpscaleImageApi = _interopRequireDefault(require("./api/UpscaleImageApi"));
var _UtilitiesApi = _interopRequireDefault(require("./api/UtilitiesApi"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
